src/js/tech/ChromecastTech.js

```diff
@@ -389,10 +432,15 @@ ChromecastTech = {
    ended: function() {
       var mediaSession = this._getMediaSession();
 
+      // Don't check for queues
+      // When handling a queue there are moments when mediaSession is null and current item has already finished
+      // and the new item is not started loading yet, which would end the session.
+      if (this._queue) {
+         return false;
```

Comment:
What if the whole queue has ended?

Reply:
I haven't found a reliable event. As in example when you are on the last chapter and you will use the chromechast control to switch to a different one the same events are fired as if the chapter would end...

Reply:
Better than nothing?..

Reply:
I think better UX is when the session stays hanging then when session is interrupted when you are attempting to watch previous episode